### PR TITLE
Implementing Time Limiting with Resilience4j

### PR DESCRIPTION
**Pull request by saajn**
Please re-read your article a day or so after you finished writing it and work through this checklist to polish the quality:

- [x] the article explains *why* and not only *how*
- [x] the text is conversational ("we" instead of "you", easy language)
- [x] there are no "walls of text" (short sentences, short paragraphs with a single idea)
- [x] the text is inclusive ("the developers" and "they" instead of "the developer" and "he/she")
- [x] the text uses active voice instead of passive voice
- [x] names are spelled consistently throughout the article (uppercase / lowercase)
- [x] the article has a short intro before the first headline
- [x] by reading the table of contents (TOC) alone, I get a sense of the "story" of the article (check the TOC in the preview after raising the PR or by starting the blog locally on your machine)
- [x] if using sub-headlines, there's at least two sub-headlines below a main-headline
- [x] (half-)sentences that express a meaningful idea that is understandable by itself own are **bold** to allow quick scanning of the text
- [x] code examples are introduced with a ":" and explained below the code example
- [x] code examples are formatted to avoid horizontal scrolling (manual line breaks and 2-spaces indentation)
- [x] the text has been checked with [Grammarly](https://grammarly.com) (the free tier is enough)

**Feel free to raise the PR even if the checklist hasn't been worked through, yet**, to take advantage of the preview that is generated for each pull request (see "deploy/netlify" under "Show all checks"). But please only flag it to be reviewed after going through the checklist above.

### CODE REVIEW

File: _posts/2020-08-08-time-limiting-with-resilience4j.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+In this series so far, we have learned about Resilience4j and its [Retry](/retry-with-resilience4j/) and [RateLimiter](/rate-limiting-with-resilience4j/) modules. In this article, we will continue exploring Resilience4j with a look into the TimeLimiter. We will find out what problem it solves, when and how to use it, and also look at a few examples.
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/resilience4j/timelimiter" %}
+
+## What is Resilience4j?
+
+Please refer to the description in the previous article for a quick intro into [how Resilience4j works in general](/retry-with-resilience4j/#what-is-resilience4j).
+
+## When to Use the TimeLimiter?
```
Comment (thombergs) [inline _posts/2020-08-08-time-limiting-with-resilience4j.md:20]: This section is very focussed on CompletableFuture.

I would like to rename this section to "Time Limiting CompletableFutures" and add another section before it "What are Time Limits for?" (or similar) explaining the concept of timeouts between client and server and why this is necessary. That section doesn't need code, just an explanation of the concept and where we find timelimits in our every day life (there are timeouts on pretty much every level of network communication).

File: _posts/2020-08-08-time-limiting-with-resilience4j.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+```
+The number of timed out calls - resilience4j.timelimiter.calls(timeout): 6.0
+The number of successful calls - resilience4j.timelimiter.calls(successful): 4.0
+The number of failed calls - resilience4j.timelimiter.calls(failed): 0.0
+```
+
+In a real application, we would export the data to a monitoring system periodically and analyze it on a dashboard.
+
+## Gotchas and Good Practices When Implementing Time Limiting
```
Comment (thombergs) [inline _posts/2020-08-08-time-limiting-with-resilience4j.md:215]: Love that explanation!

File: _posts/2020-08-08-time-limiting-with-resilience4j.md (hunk 1 above)
Comment (saajn) [reply _posts/2020-08-08-time-limiting-with-resilience4j.md]: @thombergs I have added a new section "What is Time Limiting?", please check.

I am not really sure if we should rename this to "Time Limiting CompletableFutures" though. I renamed it to "When to Use the Resilience4j TimeLimiter?" instead and also added an introductory line to the subsection, please take a look.

File: _posts/2020-08-08-time-limiting-with-resilience4j.md (hunk 2 above)
Comment (saajn) [reply _posts/2020-08-08-time-limiting-with-resilience4j.md:225]: Thanks @thombergs  :)

**Comment by saajn** [comment]
> @saajn I didn't even change a word, great article! :)

Thank you Tom! :)

**Pull request merged**